cinema/permissions.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+logger = logging.getLogger(__name__)
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    """
+    Allows read/write access for admin users (except delete),
+    and read-only access for authenticated users.

```

Comment:
The docstring mentions that admin users have read/write access except delete, but the implementation does not reflect this. Ensure that admin users are allowed to perform all actions except delete on all endpoints.